vulnerabilities/scraper/archlinux.py

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright (c) 2017 nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/vulnerablecode/
+# The VulnerableCode software is licensed under the Apache License version 2.0.
+# Data generated with VulnerableCode require an acknowledgment.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with VulnerableCode or any VulnerableCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with VulnerableCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  VulnerableCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  VulnerableCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+
+import json
+from urllib.request import urlopen
+
+
+ARCHLINUX_TRACKER_URL = 'https://security.archlinux.org/json'
+
+
+def extract_vulnerabilities(arch_data):
+    """
+    Return a sequence of mappings for each existing combination of
+    package and vulnerability from a mapping of arch linux vulnerabilities
+    data.
+    """
+    package_vulnerabilities = []
+
+    for item in arch_data:
+        if not item["name"] or not item["packages"][0] or not item["fixed"]:
+            continue
+
+        package_vulnerabilities.append({
+                'package_name': item["packages"][0],
+                'vulnerability_id': item["name"],
+                'description': item["type"],
+                'status': item["status"],
+                'severity': item["severity"],
+                'affected_version': item["affected"],
+                'fixed_version': item["fixed"]
+            })
+    return package_vulnerabilities
```

Comment:
Why not yield instead? This is not stored anywhere is it?

Reply:
@pombredanne Well I am returning it just to maintain the consistency of the methods with other implmented scrapers like ubuntu and debian.